### ISSUE TITLE
improve white box optimizers

### ISSUE THREAD

**Issue by jg10545** [jg10545/electricmayhem#110]
* abstract out iterative FGSM, other optimizers to pytorch-compatible optimizer objects
* provide optimization and learning rate decay options in pipeline training API
* options to autotune optimizer and lr decay type with `ax`

**Comment by jg10545** [comment]
closed in #113
